### PR TITLE
[feat] LSP notifications listeners support

### PR DESCRIPTION
**Pull request by rutvik110**
[feat] LSP notifications listeners support

### CODE REVIEW

**Comment by rutvik110** [comment]
@matthew-carroll Could you please enable the branch protection rules for this repo?

**Comment by matthew-carroll** [comment]
@rutvik110 which branch protection rules are you referring to?

**Comment by rutvik110** [comment]
I was talking abt the main branch protection rules, specifically these two, `Require approval before merge` and `Require all checks to pass to allow merge`. Sry, should've been clear abt that.

File: example/lib/ide/ide.dart
Comment (angelosilvestre) [inline example/lib/ide/ide.dart]: Please be sure to set the line width to 120 characters. I noticed that the formatter changed some of the existing format.

File: example/lib/ide/ide.dart
Comment (angelosilvestre) [inline example/lib/ide/ide.dart]: We don't need this parameter, because we already have a `isAnalyzing` at widget level.

**Comment by rutvik110** [comment]
@angelosilvestre Made the requested updates.

* Removed redundant method params.
* Formatting updates based on 120 line length.

**Comment by matthew-carroll** [comment]
I enabled `main` protection

**Comment by matthew-carroll** [comment]
Also @rutvik110 @suragch as a reminder, please be sure to always "Squash and merge" PRs so we don't litter the main history with a large number of little adjustment commits.

**Comment by matthew-carroll** [comment]
@angelosilvestre I'll defer to you for reviews on LSP comms. Whenever you think it's good to go, we can merge.

**Review by angelosilvestre — APPROVED** [review]
LGTM

**Pull request merged**
